src/common/cryptoutils.cpp

```diff
@@ -19,6 +19,59 @@ constexpr auto cSchemeFile      = "file";
 constexpr auto cSchemePKCS11    = "pkcs11";
 constexpr auto cSchemeMaxLength = Max(sizeof(cSchemeFile), sizeof(cSchemePKCS11));
 
+/***********************************************************************************************************************
+ * Static
+ **********************************************************************************************************************/
+
+static Error StrToPKCS11ID(const String& idStr, Array<uint8_t>& id)
+{
+    id.Clear();
+
+    auto                 percentDetected = false;
+    aos::StaticString<2> hexByte;
+
+    for (const auto& ch : idStr) {
```

Comment:
it's better to use for(int i=0; i<Size(); i +=3) to simplify the logic

Reply:
No. Current implementation is more generic and supports the following cases for example:
`id%01%12` that is not recommended but not forbidden as well.